src/core.cpp

```diff
@@ -754,4 +767,62 @@ void Core::delcore(int idx)
     // If core is null, it's already been shutdown, so do nothing. 
 }
 
+//! Retrieve the current biome list
+std::vector<std::string> Core::getBiomeList() const
+{
+    IModelComponent* snbox_i = getComponentByName( SIMPLENBOX_COMPONENT_NAME );
+    SimpleNbox* snbox = dynamic_cast<SimpleNbox*>(snbox_i);
```

Comment:
I agree that this is probably the better way to go. However, one noteworthy caveat is that I think we have to introduce code like the following to the `ocean_component.hpp`, which is _also_ a `CarbonCycleModel`:

```
    inline
    void createBiome(const std::string& biome) {
        H_THROW("Cannot call `createBiome` for ocean component.")
    }
    inline
    void deleteBiome(const std::string& biome) {
        H_THROW("Cannot call `deleteBiome` for ocean component.")
    };
    inline
    void renameBiome(const std::string& oldname, const std::string& newname){
        H_THROW("Cannot call `renameBiome` for ocean component.")
    }
```

Otherwise, we get a compile-time error because, when we go to initialize the ocean component, we end up allocating a "virtual" class, which I guess is a bad thing.

Is this an OK workaround, or are there better alternatives? Should I just implement these directly in the `CarbonCycleModel` (modifying the error slightly to say something like `createBiome` not implemented for this model)? I guess we could also consider deepening the class hierarchy such that `SimpleNbox` inherits from something like `LandCCycleModel` that in turn inherits from `CarbonCycleModel`, but that seems like overkill.

Reply:
Hmm.  I hadn't realized that the ocean component was also a `CarbonCycleModel`.  If I were going to add a new class, I'd do it using multiple inheritance rather than deepening the hierarchy.  But maybe the best course for now is just to have those methods throw a not-implemented error in the ocean component.

> Should I just implement these directly in the CarbonCycleModel (modifying the error slightly to say something like createBiome not implemented for this model)?
 
Yes, this is the best way to implement a design where the subclasses are allowed to not implement part of the interface.

Reply:
OK, I implemented this in 87665ee8b3e8169e1b23dd46c068b9d2cf241199.

Reply:
💯